### ISSUE TITLE
Unable to `include` relationships when using `relation_name` for an association.

### ISSUE THREAD

**Issue by esiegel** [venuu/jsonapi-authorization#44]
From the jsonapi-resources docs this will fail if `approved_book_comments` isn't also a relationship on the resource itself.

```ruby
has_many :book_comments, relation_name: -> (options = {}) {
      :approved_book_comments
  }
```

In `pundit_scoped_resource.rb` it should be more clever and detect that the `association_name` has been mapped to another name.  We could reevaluate the `relation_name` lambda to generate a mapping from original name to new name, but I worry that this might be expensive in some cases.

**Comment by bsofiato** [comment]
@esiegel I think this issue is addressed by the 0.8.2 version.

**Comment by esiegel** [comment]
👍
